zkevm-circuits/src/evm_circuit/execution/jumpi.rs

```diff
@@ -74,6 +63,29 @@ impl<F: Field> ExecutionGadget<F> for JumpiGadget<F> {
         };
         let same_context = SameContextGadget::construct(cb, opcode, step_state_transition);
 
+        // Lookup opcode at destination when should_jump
+        cb.condition(should_jump.clone(), |cb| {
+            cb.require_equal(
+                "JUMPI destination must be within range if condition is non-zero",
+                dest.not_overflow(),
+                1.expr(),
+            );
+
+            #[cfg(not(feature = "dual_bytecode"))]
+            cb.opcode_lookup_at(dest.valid_value(), OpcodeId::JUMPDEST.expr(), 1.expr());
```

Comment:
again these codes repeat several times. Can be refactored as a standalone function


```
#[cfg(not(feature = "dual_bytecode"))]
            cb.opcode_lookup(opcode.expr(), 1.expr());

            #[cfg(feature = "dual_bytecode")]
            {
                cb.condition(is_first_bytecode_table.expr(), |cb| {
                    cb.opcode_lookup(opcode.expr(), 1.expr());
                });
                cb.condition(not::expr(is_first_bytecode_table.expr()), |cb| {
                    cb.opcode_lookup2(opcode.expr(), 1.expr());
                });
            }
```

Reply:
this has been in `lookup_opcode` helper. and used this helper in related opcode gadgets.